Added pagination

MirgudeSaikrishna opened the pull request:
issue: 109
added pagination to the blog posts.
Displaying 10 posts per page.
made changes in posts.jsx in frontend and blogPostContrtoller.js in server.

https://github.com/user-attachments/assets/25fad598-2930-43bb-bee8-bac00378df08

MirgudeSaikrishna committed:
shows login successfull after login

MirgudeSaikrishna committed:
added pagination to blog posts

alvarotorrestx commented on client/src/assets/components/posts/Posts.jsx:
```diff
@@ -39,25 +41,27 @@ const Posts = () => {
-        fetchPosts();
-    }, [auth?.accessToken]);
+        // scrollY = window.scrollY;
+        window.scrollTo(0, 0);
```
Very nice thought here for the UX.

Any way you can make this a smooth scroll?

alvarotorrestx commented on client/src/assets/components/posts/Posts.jsx:
```diff
@@ -201,6 +205,25 @@ const Posts = () => {
             }
 
+            {/* Pagination Controls */}
+            <div className="flex justify-center mt-6">
+                <button
+                    className="btn btn-primary btn-sm"
+                    onClick={() => setCurrentPage((prev) => Math.max(prev - 1, 1))}
+                    disabled={currentPage === 1}
+                >
+                    Previous
+                </button>
+                <span className="mx-4">Page {currentPage} of {totalPages}</span>
+                <button
+                    className="btn btn-primary btn-sm"
+                    onClick={() => setCurrentPage((prev) => Math.min(prev + 1, totalPages))}
+                    disabled={currentPage === totalPages}
+                >
+                    Next
+                </button>
+            </div>
```
I really like the implementation, it works great! Is there anyway you can convert it to the styles from [DaisyUI] (https://daisyui.com/components/pagination/) as mentioned?

Here are the preferred styles:
![image](https://github.com/user-attachments/assets/900fa9b3-40bd-4a4f-9ca5-af498e86d971)
or
![image](https://github.com/user-attachments/assets/f41c4e51-bf34-45d2-8eb1-92993cbc59eb)

alvarotorrestx replied:
Also, for a better user experience, hiding the buttons when it is only 1 page would be ideal.

alvarotorrestx reviewed (changes requested):
This is a fantastic implementation! It works just as expected.

I did leave some requests and some feedback as well for improvements. Let me know if you need any further help or have concerns or feedback!

Thank you so much for your contribution! 😄

alvarotorrestx commented:
I did want to add some things I did notice to consider as well:

**Bugs:**
- When page reaches 10+ posts in same render, the dom needs a refresh to show a next page available button
- When deleting enough posts to "remove a page", meaning a page no longer has 10+ posts or the entire site does not have 10+ posts, the Next button will still show and it will take you to an empty "no posts found" page

alvarotorrestx commented:
Reference to Issue #109

MirgudeSaikrishna committed:
updated  pagination with smooth scroll and new page updation

MirgudeSaikrishna committed:
added pagination

MirgudeSaikrishna commented:
updated the program such that when a new post is added or removed the number of pages change and
also implemented the scroll to top with smooth behaviour closes: #110 
improved the UI of pagination using daisyUI.
I couldn't upload the video example due to larger size.
please review @alvarotorrestx  😊


https://github.com/user-attachments/assets/d1ce9b2f-72af-4142-bcda-2afefb592003

alvarotorrestx reviewed (approved):
Fantastic job!! 🎉

alvarotorrestx commented:
Hey great job with this @MirgudeSaikrishna 

I apologize, I am late with reviewing my PRs. I had a busy week with school. Merging this now. Thank you for your multiple contribution! 🙂

MirgudeSaikrishna commented:
No problem, thank you.
